include/zephyr/drivers/usb/emul_bc12_pi3usb9201.h

```diff
@@ -0,0 +1,20 @@
+/*
```

Comment:
Here are API headers, this header does not belong here.

Reply:
This header provides one public routine for the tests..  Do you suggest this file should be under subsys/emul/i2c/ instead (adjacent to the emulator C file)?

Reply:
@jfischer-no, since you still have a NACK on this.  Can you recommend a better location for this header?

Reply:
I suggest to place it in driver's directory, not sure about subsys/emul/i2c/, but having a header in include/zephyr/drivers/*/ for every emulated driver is not right.

Reply:
Isn't `include/zephyr/drivers/usb` the right place for this? That's the pattern used by other drivers as well for extending the subsystem APIs, see https://github.com/zephyrproject-rtos/zephyr/tree/main/include/zephyr/drivers/sensor, https://github.com/zephyrproject-rtos/zephyr/tree/main/include/zephyr/drivers/gpio etc. Haven't seen a specific guideline but I thought this was how it's meant to be done.

Reply:
I've left the emulator include file under `include/zephyr/drivers/usb/` for now.  I opened [RFC#53791](https://github.com/zephyrproject-rtos/zephyr/issues/53791) which will add support for defining backend APIs for emulators.

With that change, the file `emul_bc12_pi3usb9201.h` gets renamed to `emul_bc12.h` and can then be used with multiple BC1.2 emulators.

Reply:
> With that change, the file `emul_bc12_pi3usb9201.h` gets renamed to `emul_bc12.h` and can then be used with multiple BC1.2 emulators.

@keith-zephyr That sounds good, thanks for moving toward generic solution. Will there be this generic approach until rc1?  It would be awkward if we go into release with this specific header.

Reply:
I created https://github.com/zephyrproject-rtos/zephyr/pull/53908 to add backend API support to the emulators.  Once that PR merges, I'll update this PR to create a generate backend API for BC1.2 emulators.